examples/with-claims/api/claims/RolesClaim.ts

```diff
@@ -0,0 +1,91 @@
+import Session, { SessionClaim, SessionClaimValidator, SessionContainer } from "supertokens-node/recipe/session";
+import { roleDB } from "../mockRoleDb";
+
+class RolesClaimClass extends SessionClaim<string[]> {
+    constructor() {
+        super("st-r");
+        this.hasRole = {
+            validatorTypeId: this.key,
+            including: (role: string) => ({
+                validatorTypeId: this.key,
+                claim: this,
+                shouldRefetch: (payload, ctx) => {
+                    return this.getValueFromPayload(payload, ctx) === undefined;
+                },
+                validate: async (payload, ctx) => {
+                    return {
+                        isValid: this.getValueFromPayload(payload, ctx).includes(role),
+                        reason: { missingRole: role },
+                    };
+                },
+            }),
+        };
+    }
+    readonly hasRole: {
+        validatorTypeId: string;
+        including: (role: string) => SessionClaimValidator;
+    };
+
+    addRole(session: SessionContainer, role: string, userContext?: any) {
+        const payload = session.getAccessTokenPayload(userContext);
+        const origRoles = this.getValueFromPayload(payload, userContext);
+        const newRoles = Array.from(new Set([...origRoles, role]));
+        return session.mergeIntoAccessTokenPayload(this.addToPayload_internal({}, newRoles, userContext));
+    }
+
+    setRoles(session: SessionContainer, roles: string[], userContext?: any) {
+        return session.mergeIntoAccessTokenPayload(this.addToPayload_internal({}, roles, userContext));
+    }
+
+    updateRoles(session: SessionContainer, userContext?: any) {
```

Comment:
Consider calling this `refreshRolesFromDb` or something. It's very confusing with `addRole`